pkg/statemetrics/mr_state_metrics.go

```diff
@@ -81,18 +82,20 @@ type MRStateRecorder struct {
 	log         logging.Logger
 	interval    time.Duration
 	managedList resource.ManagedList
+	gvk         schema.GroupVersionKind
```

Comment:
I'm pretty sure you could avoid needing this argument by just trimming the `List` suffix from the `ManagedList`'s GVK.

Reply:
I'm testing the code with provider-kubernetes and in the `Record` method, `mrList`'s GVK is empty. That's why I added a separate field for `gvk` and passed it from the provider's main.

Reply:
I'm adding the recorder like below. Therefore, the metadata comes empty. I think setting the values in `&v1alpha2.ObjectList{}` is too implicit. Therefore, I'm suggesting to pass the `gvk` as a separate parameter. @negz what do you think?
```
if err := mgr.Add(statemetrics.NewMRStateRecorder(
		mgr.GetClient(), &v1alpha2.ObjectList{}, v1alpha2.ObjectGroupVersionKind, o.Logger, o.MetricOptions.MRStateMetrics, o.MetricOptions.PollStateMetricInterval)); err != nil {
		return err
	}
```

Reply:
As far as I can tell the only reason you pass `resource.ManagedList` is to let the `client.Client` know what GVK to query. Internally it uses controller-runtime's [`apiutil.GVKForObject`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.2/pkg/client/apiutil#GVKForObject) to derive a GVK from the supplied list's Go type. So to me it seems redundant to also pass the GVK explicitly.

I would suggest either:

1. Only take a type that satisfies `resource.ManagedList` as an argument, like `&v1alpha2.ObjectList{}`, and use `GVKForObject` to extract its GVK. This will require storing a `runtime.Scheme` (from `mgr.GetScheme()` in the `MRStateRecorder`.
2. Only take a GVK as an argument, and use an `UnstructuredList{}`

Example of option 2:

```go
ul := &unstructured.UnstructuredList{}
ul.SetAPIVersion(gvk.GroupVersion().String())
ul.SetKind(gvk.Kind+"List")
_ = r.client.List(ctx, ul)
```

I prefer option 1, because clients won't cache `Unstructured` types by default.

Reply:
> This will require storing a runtime.Scheme (from mgr.GetScheme() in the MRStateRecorder.

Actually, the `client.Client` has a `Scheme` method you can use to get one.

Reply:
Thanks @negz for your review. I used the `GVKForObject` to extract the GVK and removed `List` to get the object kind.